examples/platform/silabs/SiWx917/SiWx917/sl_wifi_if.cpp

```diff
@@ -559,7 +547,7 @@ sl_status_t show_scan_results(sl_wifi_scan_result_t * scan_result)
         strcpy(&ap.ssid[0], (char *) &scan_result->scan_info[x].ssid);
         if (wfx_rsi.scan_ssid)
         {
-            SILABS_LOG("SCAN SSID: %s , ap scan: %s", wfx_rsi.scan_ssid, ap.ssid);
+            ChipLogDetail(DeviceLayer, "Scan SSID: %s , AP scan: %s", wfx_rsi.scan_ssid, ap.ssid);
```

Comment:
Are you certain `ssid` is *always* nul-terminated? Looks like `wfx_wifi_scan_result_t` enforces it, but I feel a safe-guarantees-null-termination should be used for `strcpy(&ap.ssid[0], (char *) &scan_result->scan_info[x].ssid);` --> if underlying API doesn't properly nul-terminate, then this could overrun a lot, and then not guarantee null termination.

Reply:
added a null termination for this.